### PR TITLE
Reorganize option help to match /help output

### PR DESCRIPTION
**Pull request by atomb**
Reorganize the option text in the Dafny reference mangual to match the output of `/help`, and add documentation for more options. The new descriptions are mostly drawn directly from the output of `/help`, with a little editing and additional surrounding text.

Some sections are still "TO BE WRITTEN", but many fewer.

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

File: docs/DafnyRef/UserGuide.md [hunk 1]
```diff
@@ -522,53 +522,434 @@ implementation.
...
+* `-extractCounterexample` - control generation of counterexamples. If
+  verification fails, report a detailed counterexample for the first
+  failing assertion. Requires specifying the `-mv` option, to specify
+  where to write the counterexample, as well as the
+  `-proverOpt:O:model_compress=false` and
+  `-proverOpt:O:model.completion=true` options.
+
+* `-countVerificationErrors:<n>` - if 0 then always exit with a 0 exit
+  code, regardless of whether errors are found. If 1 (the default) then
+  use the usual exit code. This option is deprecated.
```
Comment (fabiomadge) [inline docs/DafnyRef/UserGuide.md:950]:
```suggestion
* `-countVerificationErrors:<n>` - if `0` then always exit with a 0 exit
  code, regardless of whether errors are found. If `1` (default) then
  use the appropriate exit code. This option is deprecated.
```

File: docs/DafnyRef/UserGuide.md [hunk 2]
```diff
@@ -522,53 +522,434 @@ implementation.
...
-* `-?` or `-help` : prints out the current list of command-line options and terminates
-* `-version` : prints the version of the executable being invoked and terminates
+These options select output including documentation on command-line
+options or attribute declarations, information on the version of Dafny
+being used, and information about how Dafny was invoked.
 
-### 24.10.2. Controlling errors and exit codes
+* `-?` or `-help` - print out the current list of command-line options
+  and terminate. All of these options are also described in this and
+  the following sections.
```
[the remark this answers is not in the document]
Comment (atomb) [reply docs/DafnyRef/UserGuide.md:538]: Oh, good point. I'll remove L526. I think it'd be good if the documentation covered all of them.

File: docs/DafnyRef/UserGuide.md [hunk 3]
```diff
@@ -522,53 +522,434 @@ implementation.
...
-### 24.10.4. Controlling aspects of the tool being run
+### 24.10.2. Controlling input {#sec-controlling-input}
 
-* `-deprecation:<n>` - controls warnings about deprecated features
+These options control how Dafny processes its input.
 
-   * 0 - no warnings
-   * 1 (default) - issue warnings
-   * 2 - issue warnings and advise about alternate syntax
+* `-dprelude:<file>` - select an alternative Dafny prelude file.
```
[the remark this answers is not in the document]
Comment (atomb) [reply docs/DafnyRef/UserGuide.md:562]: Yeah, I'll do that.

File: docs/DafnyRef/UserGuide.md [hunk 4]
```diff
@@ -522,53 +522,434 @@ implementation.
...
+    `NoIncludes` disables.
+
+* `-printIncludes:<None|Immediate|Transitive>` - select what information
+  from included files to incorporate into the output selected by
+  `-dprint` or `-rprint`. The argument can be one of the following.
+
+  * `None` (default) - don't print anything from included files.
+
+  * `Immediate` - print files directly included by files specified on
+    the command line. Exit after printing.
```
[the remark this answers is not in the document]
Comment (atomb) [reply docs/DafnyRef/UserGuide.md:612]: Agreed.

File: docs/DafnyRef/UserGuide.md [hunk 5]
```diff
@@ -522,53 +522,434 @@ implementation.
...
+  * `0` (default) - allow `assume` statements and free invariants, or
+
+  * `1` - treat all assumptions as `assert` statements, and drop free
+    invariants.
+
+* `-induction:<n>` - control the behavior of induction. The argument can
+  be one of the following.
 
-* `-verifyAllModules` - verify modules that come from include directives
+  * `0` - never do induction, not even when attributes request it.
```
[the remark this answers is not in the document]
Comment (atomb) [reply docs/DafnyRef/UserGuide.md:792]: Probably easier to change it here?

[the remark this answers is not in the document]
Comment (fabiomadge) [reply docs/DafnyRef/UserGuide.md]: That's what I thought when going top to bottom, but: L811, L823, L836, L862, L896, L930, L940, L1008, L1036, L1084

[the remark this answers is not in the document]
Comment (atomb) [reply docs/DafnyRef/UserGuide.md]: Yeah, and on further consideration, that later style is more flexible. I'll use it instead.

File: docs/DafnyRef/UserGuide.md [hunk 6]
```diff
@@ -626,31 +1030,71 @@ TO BE WRITTEN
    * `py` - Python
```
Comment (fabiomadge) [inline docs/DafnyRef/UserGuide.md:1030]: This is rendered differently because of the missing empty lines. Not sure, whether it should change.

**Review by fabiomadge — APPROVED** [review]

**Pull request merged**